### ISSUE TITLE
Prometheus Metrics for BIRD Client

### ISSUE THREAD

**Issue by arzarif** [projectcalico/calico#2369]
## Expected Behavior

Extending reported metrics via the optionally-enabled Prometheus endpoint to include BIRD-related statistics/state would improve its usefulness. 

## Current Behavior

Currently, only Felix-specific metrics can be exposed via a Prometheus endpoint:
https://docs.projectcalico.org/v3.4/reference/felix/prometheus

## Possible Solution

N/A

## Steps to Reproduce (for bugs)
N/A

## Context

Felix's Prometheus endpoint is a pretty convenient way for us to gain visibility into the health/state of calico-node, however, it obviously does not describe calico-node's state completely as the metrics are specific to the Felix daemon. It would be great if the endpoint could similarly provide metrics related to the state of the BIRD client. For example, an endpoint that could provide information pertaining to the state of connected peers for monitoring purposes would be especially helpful. 

I've seen a couple of projects that export Prometheus metrics for BIRD, however, it'd be more convenient to have this ship with calico-node itself. 

## Your Environment

N/A

**Comment by jiangdong0618** [comment]
有针对bird的metric监控实践吗？

**Comment by caseydavenport** [comment]
You might be interested in this: https://github.com/czerwonk/bird_exporter

I haven't used it, but from a glance it looks like it would do what you need.

**Comment by paketb0te** [comment]
@caseydavenport since this exporter communicates with BIRD through sockets, I guess it would make sense to deploy it in the same pod where BIRD is running, no? (If I am not mistaken, this is the calico-node pod on each cluster node)

Would you generally be open to have that exporter as an optional sidecar to enable metrics from BIRD?

**Comment by caseydavenport** [comment]
>  I guess it would make sense to deploy it in the same pod where BIRD is running, no? (If I am not mistaken, this is the calico-node pod on each cluster node)

The BIRD socket is available via a volume mount on the host, so possible it can be configured to be shared with another pod.

I think there's another level of review and analysis we'd need to do before accepting a proper integration with that project into upstream Calico, and it might result in us building something ourselves rather than relying on that project. I mostly suggested it as a potential option for anyone wanting to spin this up themselves without writing code.

**Comment by abasitt** [comment]
@caseydavenport I believe this is not available in OSS version ? 
https://docs.tigera.io/calico-cloud/operations/monitor/metrics/bgp-metrics

**Comment by caseydavenport** [comment]
Yeah, I think that's currently only implemented in the paid products.
